mollie/api/resources/base.py

```diff
@@ -0,0 +1,88 @@
+import json
+
+from ..error import RequestSetupError, ResponseError, ResponseHandlingError
+from ..objects.list import List
+
+
+class Base(object):
+    REST_CREATE = 'POST'
+    REST_UPDATE = 'PATCH'
+    REST_READ = 'GET'
+    REST_LIST = 'GET'
+    REST_DELETE = 'DELETE'
+    DEFAULT_LIMIT = 10
+
+    def __init__(self, client):
+        self.client = client
+
+    def get_resource_object(self, result):
+        raise NotImplementedError()
+
+    def get_resource_name(self):
+        return self.__class__.__name__.lower()
+
+    def rest_create(self, data, params=None):
+        path = self.get_resource_name()
+        result = self.perform_api_call(self.REST_CREATE, path, data, params)
+        return self.get_resource_object(result)
+
+    def rest_read(self, resource_id, params=None):
+        path = self.get_resource_name() + '/' + str(resource_id)
+        result = self.perform_api_call(self.REST_READ, path, params=params)
+        return self.get_resource_object(result)
+
+    def rest_update(self, resource_id, data, params=None):
+        path = self.get_resource_name() + '/' + str(resource_id)
+        result = self.perform_api_call(self.REST_UPDATE, path, data, params)
+        return self.get_resource_object(result)
+
+    def rest_delete(self, resource_id, params=None):
+        path = self.get_resource_name() + '/' + str(resource_id)
+        return self.perform_api_call(self.REST_DELETE, path, params=params)
+
+    def rest_list(self, params=None):
+        path = self.get_resource_name()
+        result = self.perform_api_call(self.REST_LIST, path, params=params)
+        return List(result, self.get_resource_object({}).__class__)
+
+    def create(self, data=None, **params):
+        if data is not None:
+            try:
+                data = json.dumps(data)
+            except Exception as e:
+                raise RequestSetupError('Error encoding parameters into JSON: "%s"' % str(e))
+        return self.rest_create(data, params)
+
+    def get(self, resource_id, **params):
+        return self.rest_read(resource_id, params)
+
+    def update(self, resource_id, data=None, **params):
+        if data is not None:
+            try:
+                data = json.dumps(data)
+            except Exception as e:
+                raise RequestSetupError('Error encoding parameters into JSON: "%s"' % str(e))
+        return self.rest_update(resource_id, data, params)
+
+    def delete(self, resource_id):
+        return self.rest_delete(resource_id)
+
+    def all(self, **params):
```

Comment:
I think we should consider renaming this to `list` or `page`, as it generally does not return all objects of a certain resource. What do you think @willemstuursma?

Reply:
That is a great idea. My vote is for `list`. That change should be part of implementing the list behaviour properly.

Reply:
LGTM

Reply:
Addressed in separate PR: https://github.com/mollie/mollie-api-python/pull/45#discussion_r208945541